firmware/Core/Src/rtos_tasks/rtos_tasks.c

```diff
@@ -163,3 +165,49 @@ void TASK_execute_telecommands(void *argument) {
 
 	} /* End Task's Main Loop */
 }
+
+void TASK_receive_gps_info(void *argument) {
+	TASK_HELP_start_of_task();
+
+	char latest_gps_response[UART_gps_buffer_len];
+	uint16_t latest_gps_response_len = 0;
+
+	while (1) {
+
+		osDelay(200);
+
+		// Checking if there is data in the GPS buffer
+		if(UART_gps_buffer_write_idx > 0) {
+
+			// Copy data from the UART buffer to a separate buffer
+			latest_gps_response_len = UART_gps_buffer_write_idx;
+			for (uint16_t i = 0; i < UART_gps_buffer_len; i++) {
+				latest_gps_response[i] = (char) UART_gps_buffer[i];
+			}
+
+			// Set the null terminator at the end of the `latest_gps_response` str.
+			latest_gps_response[latest_gps_response_len] = '\0';
+
+			// Clear the buffer (memset to 0, but volatile-compatible) and reset the write pointer.
+			for (uint16_t i = 0; i < UART_gps_buffer_len; i++) {
+				UART_gps_buffer[i] = 0;
+			}
+			UART_gps_buffer_write_idx = 0;
+
+		}
+
+		// Parsing the gps data
+		gps_response_header gps_header;
+		const u_int8_t gps_header_result = parse_gps_header(latest_gps_response,&gps_header);
+
+		// TODO: Figure out what to do after this
+		// Parse may have failed due to incomplete data
+		// How do I handle the incomplete data? Could drop the data
+		if(gps_header_result != 0){
+			// Failed to parse
+			break;
```

Comment:
Breaking this loop ends the task, forever (until the next reboot). I wouldn't do that.

Reply:
Sure thing, I replaced it with a `return` instead for now.